siad/server.go

```diff
@@ -366,6 +372,29 @@ func (srv *Server) daemonHandler(password string) http.Handler {
 	return router
 }
 
+// memloggingGET returns a json response containing a bool that tells whether or
+// not memlogging is active or not.
+func (srv *Server) memloggingGET(w http.ResponseWriter, req *http.Request, ps httprouter.Params) {
+	api.WriteJSON(w, MemloggingInfo{Active: build.MEMLOGGING})
+}
+
+// memloggingPOST makes a POST request with a boolean param `active` that sets
+// memlogging on if true, off if false.
+func (srv *Server) memloggingPOST(w http.ResponseWriter, req *http.Request, _ httprouter.Params) {
+	val := strings.ToLower(req.FormValue("active")) // response is case-insensitive
+	if val == "true" {
+		build.MEMLOGGING = true
+		api.WriteSuccess(w)
+		return
+	}
+	if val == "false" {
+		build.MEMLOGGING = false
```

Comment:
This is going to cause a race condition, because you are writing to this variable outside of a lock, and then reading it in the loggers inside of a lock. Cleanest way to avoid this I think is to just use atomics, and then instead of having the build variable be a bool, have it be a 1 or a 0.

The package for atomics is `sync/atomic`. Convention requires you prefix the name with the word 'atomic', so the global variable name becomes `AtomicMemLogging`. Use `atomic.StoreUint64` and `atomic.LoadUint64` to read and set the value for the memlogging.